From 39238796be4a020dbc02a2790db1102a0337e3c0 Mon Sep 17 00:00:00 2001
From: Pau Ruiz Safont <pau.safont@vates.tech>
Date: Fri, 8 May 2026 17:10:27 +0100
Subject: [PATCH] record_util: move to a new private library

This module can help with printing API types, so move it to a new
private library so it can be used without compiling the cli server. This
is especially interesting for unit tests.

Signed-off-by: Pau Ruiz Safont <pau.safont@vates.tech>
---
 ocaml/tests/dune             |  2 ++
 ocaml/tests/record_util/dune | 16 ++++++++++++----
 ocaml/xapi-cli-server/dune   |  8 ++++++++
 ocaml/xapi/dune              |  2 ++
 4 files changed, 24 insertions(+), 4 deletions(-)

diff --git a/ocaml/tests/dune b/ocaml/tests/dune
index f829e72c8..7ee0ec158 100644
--- a/ocaml/tests/dune
+++ b/ocaml/tests/dune
@@ -23,6 +23,7 @@
     mirage-crypto
     mtime
     pam
+    record_util
     result
     rpclib.core
     rpclib.json
@@ -99,6 +100,7 @@
     pam
     ptime
     result
+    record_util
     rpclib.core
     rpclib.json
     rresult
diff --git a/ocaml/tests/record_util/dune b/ocaml/tests/record_util/dune
index a91a104da..0bce8316d 100644
--- a/ocaml/tests/record_util/dune
+++ b/ocaml/tests/record_util/dune
@@ -1,6 +1,14 @@
 (test
-	(name test_record_util)
-	(package xapi)
-	(libraries alcotest xapi_cli_server rpclib.core xapi_consts xapi_types astring fmt)
-	(action (run %{test} --show-errors))
+  (name test_record_util)
+  (package xapi)
+  (libraries
+    alcotest
+    astring
+    fmt
+    record_util
+    rpclib.core
+    xapi_consts 
+    xapi_types
+    )
+  (action (run %{test} --show-errors))
 )
diff --git a/ocaml/xapi-cli-server/dune b/ocaml/xapi-cli-server/dune
index fb2a713dc..992d81617 100644
--- a/ocaml/xapi-cli-server/dune
+++ b/ocaml/xapi-cli-server/dune
@@ -8,9 +8,16 @@
     (run %{gen} utils --filter-internal --filter closed)))
 )
 
+(library
+  (name record_util)
+  (modules generated_record_utils record_util)
+  (libraries rpclib.core xapi-consts xapi-types)
+)
+
 (library
   (name xapi_cli_server)
   (modes best)
+  (modules (:standard \ generated_record_utils record_util))
   (libraries
     astring
     base64
@@ -19,6 +26,7 @@
     rpclib.core
     rpclib.xml
     re
+    record_util
     result
     rresult
     sexplib
diff --git a/ocaml/xapi/dune b/ocaml/xapi/dune
index 139bf4d0f..b2c78a30c 100644
--- a/ocaml/xapi/dune
+++ b/ocaml/xapi/dune
@@ -170,6 +170,7 @@
   rpclib.json
   rpclib.xml
   re
+  record_util
   result
   rresult
   rrd-transport.lib
@@ -297,6 +298,7 @@
   forkexec
   http_lib
   httpsvr
+  record_util
   rpclib.core
   rpclib.json
   rpclib.xml
